core/state-machine/src/testing.rs

```diff
@@ -118,7 +130,8 @@ impl<H: Hasher<Out=H256>, N: ChangesTrieBlockNumber> TestExternalities<H, N> {
 	///
 	/// Returns the result of the given closure.
 	pub fn execute_with<R>(&mut self, execute: impl FnOnce() -> R) -> R {
-		externalities::set_and_run_with_externalities(self, execute)
+		let mut ext = self.ext();
+		externalities::set_and_run_with_externalities(&mut ext, execute)
 	}
```

Comment:
ext is dropped, does it contains some value that are thus discarded ?

the fields: `storage_transaction` and `changes_trie_transaction` are not kept after this execute_with, this is a change, does it has some consequences ?

Reply:
As we are not really storing `TestExternalities` in the database, I think this is fine.

Reply:
It is without consequence at this point (it was one of the main difference between both implementation), but it could be good to add those to the test externalities struct (currently it means there was probably no low level test on this memoization)).

Reply:
> (currently it means there was probably no low level test on this memoization)).

Or not between multiple call to execute_with.